databricks/sdk/mixins/compute.py

```diff
@@ -21,9 +21,9 @@ class SemVer:
     build: Optional[str] = None
 
     # official https://semver.org/ recommendation: https://regex101.com/r/Ly7O1x/
-    # with addition of "x" wildcards for minor/patch versions
+    # with addition of "x" wildcards for minor/patch versions. Also, patch version may be omitted.
     _pattern = re.compile(r"^"
-                          r"(?P<major>0|[1-9]\d*)\.(?P<minor>x|0|[1-9]\d*)\.(?P<patch>x|0|[1-9x]\d*)"
+                          r"(?P<major>0|[1-9]\d*)\.(?P<minor>x|0|[1-9]\d*)(\.(?P<patch>x|0|[1-9x]\d*))?"
```

Comment:
I understand we make patch version optional so it works for snapshot (staging) but won't this make the semver less strict for production use cases?

Reply:
we can always check on prod :)

Reply:
Do you think there is a risk there?

Reply:
I tried this in prod, and it worked there as well.